File: nextcart_admin_frontend/src/services/categoryService.js
import apiClient from './apiClient'
import { ENDPOINTS } from '@/constants/api'

const categoryService = {
  getCategories: async () => {
    const response = await apiClient.get(ENDPOINTS.CATEGORIES)
    return response.data ?? []
  },

  createCategory: async (payload) => {
    const response = await apiClient.post(ENDPOINTS.CATEGORIES, payload)
    return response.data
  },

  updateCategory: async (categoryId, payload) => {
    const response = await apiClient.put(`${ENDPOINTS.CATEGORIES}/${categoryId}`, payload)
    return response.data
  },

  deleteCategory: async (categoryId) => {
    const response = await apiClient.delete(`${ENDPOINTS.CATEGORIES}/${categoryId}`)
    return response.data
  },
}

export default categoryService

File: nextcart_backend/src/main/java/com/mhjoy/nextcart/product/service/CategoryService.java
package com.mhjoy.nextcart.product.service;

import com.mhjoy.nextcart.product.dto.request.CreateCategoryRequestDto;
import com.mhjoy.nextcart.product.dto.request.UpdateCategoryRequestDto;
import com.mhjoy.nextcart.product.dto.response.CategoryResponseDto;

import java.util.List;

public interface CategoryService {

    CategoryResponseDto createCategory(CreateCategoryRequestDto requestDto);

    CategoryResponseDto updateCategory(Long categoryId, UpdateCategoryRequestDto requestDto);

    void deleteCategory(Long categoryId);

    List<CategoryResponseDto> getCategories();
}
